Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean

```diff
@@ -62,8 +62,8 @@ lemma IsTotallyUnimodular.apply {A : Matrix m n R} (hA : A.IsTotallyUnimodular)
   convert hA 1 f g (Function.injective_of_subsingleton f) (Function.injective_of_subsingleton g)
   exact (det_fin_one (A.submatrix f g)).symm
 
-lemma IsTotallyUnimodular.submatrix {A : Matrix m n R} (hA : A.IsTotallyUnimodular) {k : ℕ}
-    (f : Fin k → m) (g : Fin k → n) :
+lemma IsTotallyUnimodular.submatrix {A : Matrix m n R} (f : m' → m) (g : n' → n)
+    (hA : A.IsTotallyUnimodular) :
```

Comment:
This is a similar "Fin k -> Fintype" change as was discussed elsewhere, except that in this case we don't even need it to be a fintype!

Reply:
Perhaps more crucially, you also don't need the two types to be the same.